src/ui/public/chrome/directives/app_switcher_link/app_switcher_link.less

```diff
@@ -22,18 +23,19 @@
       text-decoration: none;
     }
   }
+
+  .is-app-links-wrapper-open & {
+    width: @as-open-width;
+  }
```

Comment:
Yes, it could... when I first saw this kind of "pattern", I also had a "woah, weird" reaction. After seeing it a few times I'm used to it now, and I also appreciate how it keeps the state grouped with the component.

I just tried pulling it out and seeing how it looked:

```less
.app-switcher-link {
  position: relative;
  width: @as-closed-width;
  height: @app-icon-height;
  line-height: @app-line-height;

  &.is-app-switcher-link-disabled {
    opacity: 0.5;

    .app-switcher-link__anchor {
      cursor: default;
    }
  }

  &:hover,
  &.active {
    background-color: @app-links-active-background;

    .app-switcher-link__anchor {
      color: @white;
      text-decoration: none;
    }
  }
}

.is-app-links-wrapper-open {
  .app-switcher-link {
    width: @as-open-width;
  }
}
```

It definitely becomes a little less readable to me. Mind if we keep it as-is?